src/lang/window.js

```diff
@@ -22,6 +22,19 @@
      */
     window.me = window.me || {};
 
+    /**
+    * Empty BaseClass constructor and prototype attached to 
+    * `me` global object to act as the base class for inheritance 
+    */
+    me.BaseClass = function () {};
+    me.BaseClass.prototype = Object.create(Object);
```

Comment:
To me, it made sense to separate the two. It's really not necessary, though.
1. Define the `extend` property on a `BaseClass` for simple 'class' inheritance - so everything inherits from `me.BaseClass.prototype`.  
2. Assign a `new BaseClass()` to `me.Object` so that `me.Object.extend()` could be used in place of `Object.extend()`.  Doesn't have to be assigned to `me.Object` either, but I just thought that would be the easiest transition from `Object.extend()`.